web/src/app/modules/shared/components/abstract-view/abstract-view.component.ts

```diff
@@ -4,10 +4,12 @@ import {
   Input,
   OnInit,
   Output,
+  Directive,
 } from '@angular/core';
 import { View } from '../../models/content';
 
-export abstract class AbstractViewComponent<T>
+@Directive()
```

Comment:
Yeah, that's a weird one.  It was actually added automatically during `ng update` migration. Basically, compiler is more strict now in enforcing that Angular classes are properly decorated. It just wouldn't compile that code without a decorator and apparently it would not be able to instantiate objects dynamically during runtime.  

I tried to use `@Component` but that requires decoration on base class with identical selectors, so it defies the purpose of abstract classes. That's why they use `@Directive` decorator which is decorator for abstract class as explained in migration notes:

```The presence of the @Directive decorator causes Angular to generate extra code for the affected class. If that decorator includes no properties (metadata), the directive won't be matched to elements or instantiated directly, but other classes that extend the directive class will inherit this generated code. You can think of this as an "abstract" directive.```

There is more info about this migration rule [here](https://angular.io/guide/migration-undecorated-classes)

Reply:
That's interesting and good to know.